api/wb/views.py

```diff
@@ -0,0 +1,95 @@
+from rest_framework import status
+from rest_framework.exceptions import ValidationError, NotFound
+from rest_framework.response import Response
+
+from osf.models import AbstractNode
+from rest_framework.views import APIView
+from addons.osfstorage.models import OsfStorageFileNode, OsfStorageFolder
+from api.base.utils import get_object_or_error
+from api.base.parsers import HMACSignedParser
+from api.wb.serializers import (
+    WaterbutlerMetadataSerializer
+)
+
+class FileMetadataView(APIView):
+    """
+    Mixin with common code for WB move/copy hooks
+    """
+    parser_classes = (HMACSignedParser,)
+    serializer_class = WaterbutlerMetadataSerializer
+    view_category = 'wb'
+    node_lookup_url_kwarg = 'node_id'
+
+    def get_object(self):
+        return self.get_node(self.kwargs[self.node_lookup_url_kwarg])
+
+    def get_node(self, node_id):
+        node = get_object_or_error(
+            AbstractNode,
+            node_id,
+            self.request,
+            display_name='node'
+        )
+        if node.is_registration and not node.archiving:
+            raise ValidationError('Registrations cannot be changed.')
+        return node
+
+    def get_serializer_context(self):
+        """
+        Extra context provided to the serializer class.
+        """
+        return {
+            'view': self
+        }
+
+    def post(self, request, *args, **kwargs):
+        serializer = self.serializer_class
```

Comment:
You'll still need to instantiate the serializer like you were doing before, except without the hardcoded class:

```python
serializer = self.serializer_class(....)
```

Reply:
Made this change in deaa1a4

Reply:
thank you for fixing for me @sloria!